test/modules/TestResultCoordinator/Reports/DirectMethod/LongHaul/DirectMethodLongHaulReport.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator.Reports.DirectMethod.LongHaul
+{
+    using System;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    class DirectMethodLongHaulReport : TestResultReportBase
+    {
+        public DirectMethodLongHaulReport(
+            string testDescription,
+            string trackingId,
+            string senderSource,
+            Option<string> receiverSource,
+            string resultType,
+            ulong senderSuccesses,
+            Option<ulong> receiverSuccesses,
+            ulong statusCodeZero,
+            ulong unknown)
+            : base(testDescription, trackingId, resultType)
+        {
+            if (receiverSource.HasValue ^ receiverSuccesses.HasValue)
+            {
+                throw new ArgumentException("Provide both receiverSource and receiverSuccesses or neither.");
+            }
+
+            this.SenderSource = Preconditions.CheckNonWhiteSpace(senderSource, nameof(senderSource));
+            this.ReceiverSource = receiverSource;
+            this.SenderSuccesses = senderSuccesses;
+            this.ReceiverSuccesses = receiverSuccesses;
+            this.StatusCodeZero = statusCodeZero;
+            this.Unknown = unknown;
+        }
+
+        public string SenderSource { get; }
+        public Option<string> ReceiverSource { get; }
+        public ulong SenderSuccesses { get; }
+        public Option<ulong> ReceiverSuccesses { get; }
+        public ulong StatusCodeZero { get; }
+        public ulong Unknown { get; }
+
+        public override string Title => this.ReceiverSource.HasValue ?
+            $"DirectMethod LongHaul Report for [{this.SenderSource}] and [{this.ReceiverSource.OrDefault()}] ({this.ResultType})" : $"DirectMethod Report for [{this.SenderSource}] ({this.ResultType})";
+
+        public override bool IsPassed => this.IsPassedHelper();
+
+        public bool IsPassedHelper()
+        {
+            bool noUnknownResults = this.Unknown > 0 ? false : true;
```

Comment:
I think it can be more readable to fail immediately if there are unknown, then remove the unknown results from the below logic.